test/e2e/prometheusagent_test.go

```diff
@@ -204,3 +209,81 @@ func testPrometheusAgentStatusScale(t *testing.T) {
 		t.Fatalf("expected scale of 2 shards, got %d", pAgent.Status.Shards)
 	}
 }
+
+func testPromAgentDaemonSetResourceUpdate(t *testing.T) {
+	//t.Parallel()
+	ctx := context.Background()
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+
+	ns := framework.CreateNamespace(ctx, t, testCtx)
+	framework.SetupPrometheusRBAC(ctx, t, testCtx, ns)
+	_, err := framework.CreateOrUpdatePrometheusOperatorWithOpts(
+		ctx, testFramework.PrometheusOperatorOpts{
+			Namespace:           ns,
+			AllowedNamespaces:   []string{ns},
+			EnabledFeatureGates: []string{"PrometheusAgentDaemonSet"},
+		},
+	)
+	require.NoError(t, err)
+
+	name := "test"
+	p := framework.MakeBasicPrometheusAgentDaemonSet(ns, name)
+
+	p.Spec.Resources = v1.ResourceRequirements{
+		Requests: v1.ResourceList{
+			v1.ResourceMemory: resource.MustParse("100Mi"),
+		},
+	}
+
+	p, err = framework.CreatePrometheusAgentAndWaitUntilReady(context.Background(), ns, p)
+	require.NoError(t, err)
+
+	pods, err := framework.KubeClient.CoreV1().Pods(ns).List(ctx, pa.ListOptions(name))
+	require.NoError(t, err)
+
+	res := pods.Items[0].Spec.Containers[0].Resources
+	require.Equal(t, res, p.Spec.Resources)
```

Comment:
Not sure what's the goal when listing pods here, could you explain?

Reply:
To verify that the resource settings are propagated to the pods? But I suggest that you check the daemonset directly.